File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stk=[]
    
    for ch in s:
        if ch=="(":
            stk.append(ch)
        else:
            if stk and stk[-1]=="(":
                stk.pop()
            else:
                stk.append(ch)
    if stk:
        answer=False
    
    return answer

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    boolean solution(String s) {
        boolean answer=true;
        
        Stack<String> sk=new Stack<>();
        for(int i=0;i<s.length();i++){
           if(s.charAt(i)=='('){
               sk.push(Character.toString(s.charAt(i)));
               answer=true;
           }
            else{
                if(sk.isEmpty()|| sk.pop()==Character.toString(s.charAt(i)))
                    return false;
            }
        }

        return sk.isEmpty();
    }
}